src/Validation/Rules/KnownArgumentNamesRule.hack

```diff
@@ -0,0 +1,25 @@
+namespace Slack\GraphQL\Validation;
+
+
+final class KnownArgumentNamesRule extends ValidationRule {
+    <<__Override>>
+    public function enter(\Graphpinator\Parser\Node $node): void {
+        if ($node is \Graphpinator\Parser\Value\ArgumentValue) {
```

Comment:
> I had to add back the Graphpinator\Parser\Value\ArgumentValue AST node since this rule should only execute against arguments, not arbitrary value nodes. @jjergus , let me know if there's a better way to do this.

You could do something like

```Hack
if ($node is Parser\Field) {
  foreach ($node->getArguments() as $name => $value) {
    ...
```

here but your solution is good too. I only removed it so that I could reuse the same coercion method for arguments and input object fields -- and it looks like you addressed that by adding `getArgumentValues`.

Reply:
Yeah, I think it's nice for the validation rules to not need to do too much AST traversal themselves. Since then they'd need to manage keeping `TypeInfo` in sync too.